service/worker/scheduler/workflow.go

```diff
@@ -810,20 +813,32 @@ func (s *scheduler) startWorkflow(
 		LastCompletionResult: s.State.LastCompletionResult,
 		ContinuedFailure:     s.State.ContinuedFailure,
 	}
-	var res schedspb.StartWorkflowResponse
-	err := workflow.ExecuteLocalActivity(ctx, s.a.StartWorkflow, req).Get(s.ctx, &res)
-	if err != nil {
-		return nil, err
-	}
+	for {
+		var res schedspb.StartWorkflowResponse
+		err := workflow.ExecuteLocalActivity(ctx, s.a.StartWorkflow, req).Get(s.ctx, &res)
```

Comment:
if this keep happening, will this blow up the history size?

Reply:
It happens at most once, that's what CompletedRateLimitSleep is for